extensions/notebook/src/test/protocol/notebookUriHandler.test.ts

```diff
@@ -107,9 +114,47 @@ describe('Notebook URI Handler', function (): void {
 				.reply(httpErrorCode);
 
 			await notebookUriHandler.handleUri(notebookUri);
-			
 			sinon.assert.callCount(showErrorMessageSpy, 1);
 			sinon.assert.notCalled(showNotebookDocumentStub);
 		});
 	});
+
+	it('should open notebook when file uri is valid', async function (): Promise<void> {
+		let showQuickPickStub = sinon.stub(vscode.window, 'showQuickPick').resolves(Promise.resolve(loc.msgYes) as any);
+		let notebookDir: string = path.join(os.tmpdir(),'notebook');
+		let notebookPath: string = path.join(notebookDir,'hello.ipynb');
+		// Need to remove temp file if it exists
+		if(await exists(notebookDir)){
```

Comment:
This can cause problems if you're running tests in parallel or multiple tests are doing something similar. It's strongly suggested to generate a random folder to use for each test run instead of a hardcoded one